CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+#
+# CMakeLists file for OpenSSL
+# Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+
+cmake_minimum_required(VERSION 3.17)
```

Comment:
Is there any strong reason for so low CMake version requirement?
CMake 3.19 introduced [cmake presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) which can be used to provide sets of given configurations (replacing what Configure file does now?)
CMake 3.20 introduced [cmake_path](https://cmake.org/cmake/help/latest/command/cmake_path.html) that has some really handy path operations that are useful when you need to call cross-os external scripts using native paths.
As far as I know, Ubuntu 22.04.1 LTS now supports CMake 3.22.1. I don't know how other Linux distros but you can always install the standalone newest CMake from their install scripts.
I would really consider using CMake 3.22 if there are no other objections against it.

Reply:
Ubuntu 20.04: cmake 3.16.3 plus patches
Debian 11 (stable): cmake 3.18.4

Yeah I know what you said, "you can always install...".  Still...

Reply:
Using very recent cmake versions make the usefulness of any cmake support pretty irrelevant. 
I'd suggest that 3.17 is actually too recent to be practically useful - Ubuntu 20.04 shipped with 3.16.3 for reference - so you wouldn't even be supporting something as recent as that with this update. Going for even later versions than 3.17 would reduce its usefulness even further.